legacy-fabric-command-api-v2/1.6.4/src/main/java/net/legacyfabric/fabric/api/command/v2/Selector.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2020 - 2024 Legacy Fabric
+ * Copyright (c) 2016 - 2022 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.legacyfabric.fabric.api.command.v2;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Sets;
+
+import net.minecraft.command.CommandSource;
+import net.minecraft.entity.Entity;
+import net.minecraft.server.MinecraftServer;
+
+public enum Selector {
+	ALL_ENTITIES('e') {
+		@Override
+		public Set<Entity> resolve(CommandSource sender) {
+			return Sets.newHashSet(sender.getWorld().entities);
+		}
+	},
+	ALL_PLAYERS('a') {
+		@Override
+		public Set<Entity> resolve(CommandSource sender) {
+			return (Set<Entity>) sender.getWorld().playerEntities.stream().map(e -> (Entity) e).collect(Collectors.toSet());
+		}
+	},
+	NEAREST_PLAYER('p') {
+		@Override
+		public Set<Entity> resolve(CommandSource sender) {
+			return Sets.newHashSet(sender.getWorld().getClosestPlayer(sender.method_4086().x, sender.method_4086().y, sender.method_4086().z, 50.0D));
+		}
+	},
+	RANDOM_PLAYER('r') {
+		@Override
+		public Set<Entity> resolve(CommandSource sender) {
+			try {
+				return Sets.newHashSet((Entity) MinecraftServer.getServer().getPlayerManager().players.stream().findAny().orElseThrow(NullPointerException::new));
+			} catch (Throwable e) {
+				throw new RuntimeException(e);
+			}
+		}
+	},
+	EXECUTING_ENTITY('s') {
+		@Override
+		public Set<Entity> resolve(CommandSource sender) {
+			//TODO @s didn't exist this early in the game's development, and there seems to be no code to handle it, so maybe this'll work?
+			return Sets.newHashSet(sender.getWorld().getPlayerByName(sender.getUsername()));
```

Comment:
is there no other way to get the entity? doesn't CommandSource expose an Entity? `@s` didn't exist in 1.8.9 either